custom/aaa/models.py

```diff
@@ -405,6 +426,293 @@ def agg_from_household_case_ucr(cls, domain, window_start, window_end):
+        base_tablename = Woman._meta.db_table
+
+        return """
+        INSERT INTO "{agg_tablename}" AS agg (
+            domain, {location_levels}, month,
+            registered_eligible_couples,
+            registered_pregnancies,
+            eligible_couples_using_fp_method
+        ) (
+            SELECT
+                %(domain)s, {location_levels}, %(window_start)s AS month,
+                COUNT(*) FILTER (WHERE
+                    (NOT daterange(%(window_start)s, %(window_end)s) && any(pregnant_ranges))
+                    AND migration_status IS DISTINCT FROM 'migrated'
+                    AND marital_status = 'married'
+                ) as registered_eligible_couples,
+                COUNT(*) FILTER (WHERE
+                    daterange(%(window_start)s, %(window_end)s) && any(pregnant_ranges)
+                    AND migration_status IS DISTINCT FROM 'migrated'
+                ) as registered_pregnancies,
+                COUNT(*) FILTER (WHERE
+                    daterange(%(window_start)s, %(window_end)s) && any(fp_current_method_ranges)
+                    AND migration_status IS DISTINCT FROM 'migrated'
+                ) as eligible_couples_using_fp_method
+            FROM "{woman_tablename}" woman
+            WHERE daterange(opened_on, closed_on) && daterange(%(window_start)s, %(window_end)s)
+            GROUP BY {location_levels}
+        )
+        ON CONFLICT ({location_levels}, month) DO UPDATE SET
+           registered_eligible_couples = EXCLUDED.registered_eligible_couples,
+           registered_pregnancies = EXCLUDED.registered_pregnancies,
+           eligible_couples_using_fp_method = EXCLUDED.eligible_couples_using_fp_method
+        """.format(
+            agg_tablename=cls._meta.db_table,
+            woman_tablename=base_tablename,
+            location_levels=', '.join(cls.location_levels),
+        ), {'domain': domain, 'window_start': window_start, 'window_end': window_end}
+
+    @classmethod
+    def agg_from_ccs_record_table(cls, domain, window_start, window_end):
+        base_tablename = CcsRecord._meta.db_table
+
+        return """
+        INSERT INTO "{agg_tablename}" AS agg (
+            domain, {location_levels}, month,
+            high_risk_pregnancies,
+            institutional_deliveries,
+            total_deliveries
+        ) (
+            SELECT
+                %(domain)s, {location_levels}, %(window_start)s AS month,
+                COUNT(*) FILTER (WHERE hrp = 'yes') as high_risk_pregnancies,
+                COUNT(*) FILTER (WHERE child_birth_location = 'hospital') as institutional_deliveries,
+                COUNT(*) FILTER (WHERE
+                    child_birth_location IS NOT NULL OR child_birth_location != ''
+                ) as total_deliveries
+            FROM "{woman_tablename}" woman
+            WHERE daterange(opened_on, closed_on) && daterange(%(window_start)s, %(window_end)s) AND
```

Comment:
@calellowitz @Rohit25negi FYI that I found using dateranges to make date filters much more readable. You may want to consider using these in future dashboard queries https://www.postgresql.org/docs/9.6/functions-range.html#RANGE-OPERATORS-TABLE